addons/website_blog/views/website_blog_templates.xml

```diff
@@ -320,12 +320,12 @@ list of filtered posts (by date or tag).
                                 <t t-set="hide_date" t-value="True"/>
                             </t>
                         </span>
-                        <span t-if="len(blog_post.message_ids) > 0" class="text-nowrap ps-2 o_not_editable">|
+                        <span t-if="len(blog_post.message_ids.filtered(lambda r: r.body != '')) > 0" class="text-nowrap ps-2 o_not_editable">|
                             <i class="fa fa-comment text-muted me-1"/>
                             <a href="#discussion">
-                                <t t-esc="len(blog_post.message_ids)"/>
-                                <t t-if="len(blog_post.message_ids)>1">Comments</t>
-                                <t t-else="">Comment</t>
+                                <t t-esc="len(blog_post.message_ids.filtered(lambda r: r.body != ''))"/>
```

Comment:
Are some messages likely to be empty for some reason? Otherwise doesn't seem necessary to filter

Reply:
Deleted messages are still available in the database with an empty string in the body field.

Reply:
Deleting from backend by removing content ? Indeed, but that's not the way to remove messages ... you can publish / unpublish comments normally ? Not an expert of blog so not sure the option still exists.

Reply:
@tde-banana-odoo no that's not what this fix is doing. This fix is intended to display the correct amount of comments. Without this fix you'd see that deleted messages are also counted. The counter is displaying an incorrect value. 

Comments are deleted on the blog. (As a developer trying to solve this issue), I queried the Comments table and found out that the comments which are supposed to be deleted, are still available in the Comments table.

Reply:
@iTassisto I think the question is, why is deleting a comment represented by an empty body? The comment should be archived, if not unlinked.

The issue then is that the comments are left as empty messages instead of properly removed.

Reply:
Closing some old tabs @tde-banana-odoo 

Turns out, that's how discuss deletes messages from the front-end, seemingly to keep notifications alive somehow.

Seems like it could make sense to have this as a fix at least in stable. Would need to take into account that some messages may only have an attachment and no body, etc...

See discuss PR in which I mention this one, below.

Reply:
It makes no sense, and I won't fix every addon because they decided to do random things with models. If we decide to delete messages, they should be removed, not voided. Otherwise we end up with filtering everywhere in the code that means nothing (and are costly).

Reply:
Just jumping in, code owner notification got me interested.

> It makes no sense

Seems very louche indeed :grimacing: 

> and I won't fix every addon because they decided to do random things with models

"every addon", you mean discuss? (If I got it correctly reading @reth-odoo explanation)

Because right now, testing this quickly (got curious reading the exchange), it seems to impact the same way all frontend modules having a frontend chatter, including slides (in `website_*` modules it seems that event exibitor/event track/multiple forum models are also inheriting the `mail.thread` mixin, not sure if those got a frontend chatter tho, did not check, and probably other models too which are used in `website_*` module but got the `mail.thread` inheritance in module not depending of website)

![image](https://github.com/odoo/odoo/assets/30048408/12258b69-94fc-434e-8274-022cf4c85609)

Deleting messages on the backend record seems louche tho, people should unpublish it instead imo, but since we allow them to do it, it kind of make sense to remove spam or agressive stuff or whatever.
It's at least worth notifying the discuss team about this behavior? Maybe they can improve that in master at least.